spark/src/main/scala/geotrellis/spark/io/index/MergeQueue.scala

```diff
@@ -25,128 +27,54 @@ object MergeQueue{
-      val result = array.clone
-      System.arraycopy(array, 0, result, 0, i)
-      System.arraycopy(array, i, result, i + 1, _size - i)
-      result(i) = range
-      array = result
-    }
-    _size += 1
-  }
- 
- 
-  /** Ensure that the internal array has at least `n` cells. */
-  protected def ensureSize(n: Int) {
-    // Use a Long to prevent overflows
-    val arrayLength: Long = array.length
-    if (n > arrayLength - 1) {
-      var newSize: Long = arrayLength * 2
-      while (n > newSize) {
-        newSize = newSize * 2
-      }
-      // Clamp newSize to Int.MaxValue
-      if (newSize > Int.MaxValue) newSize = Int.MaxValue
- 
-      val newArray: Array[(Long, Long)] = new Array(newSize.toInt)
-      scala.compat.Platform.arraycopy(array, 0, newArray, 0, _size)
-      array = newArray
-    }
+
+private class RangeComparator extends java.util.Comparator[(Long, Long)] {
+
+  def compare(r1: (Long, Long), r2: (Long, Long)): Int = {
+    val retval = (r1._2 - r2._2)
+    if (retval < 0) +1
+    else if (retval == 0) 0
+    else -1
   }
- 
-  val ordering = implicitly[Ordering[(Long, Long)]]
- 
-  /** Inserts a single range into the priority queue.
-   *
-   *  @param  range        the element to insert.
-   */
-  @annotation.tailrec
-  final def +=(range: (Long, Long)): Unit = {
-    val res = if(_size == 0) { -1 } else { java.util.Arrays.binarySearch(array, 0, _size, range, ordering) }
-    if(res < 0) {
-      val i = -(res + 1)
-      var (thisStart, thisEnd) = range
-      var removeLeft = false
- 
-      var removeRight = false
-      var rightRemainder: Option[(Long, Long)] = None
- 
-      // Look at the left range
-      if(i != 0) {
-        val (prevStart, prevEnd) = array(i - 1)
-        if(prevStart == thisStart) {
-          removeLeft = true
-        }
-        if (prevEnd + 1 >= thisStart) { 
-          removeLeft = true
-          thisStart = prevStart
-          if(prevEnd > thisEnd) {
-            thisEnd = prevEnd
-          }
-        }
-      }
- 
-      // Look at the right range
-      if(i < _size  && _size > 0) {
-        val (nextStart, nextEnd) = array(i)
-        if(thisStart == nextStart) {
-          removeRight = true
-          thisEnd = nextEnd
-        } else {
-          if(thisEnd + 1 >= nextStart) {
-            removeRight = true
-            if(nextEnd - 1 >= thisEnd) {
-              thisEnd = nextEnd
-            } else if (nextEnd < thisEnd - 1) {
-              rightRemainder = Some((nextEnd + 1, thisEnd))
-              thisEnd = nextEnd
-            }
-          }
-        }
-      }
- 
-      if(removeRight) { 
-        if(!removeLeft) {
-          array(i) = (thisStart, thisEnd)
-        } else {
-          array(i-1) = (thisStart, thisEnd)
-          removeElement(i)
+}
+
+class MergeQueue(initialSize: Int = 1) {
+  private val treeSet = new java.util.TreeSet(new RangeComparator()) // Sorted data structure
+  private val fudge = 1 // Set this to one to preserve original behavior
```

Comment:
What does this mean? What is the original behavior vs the non-original behavior, and why did it change?

Consider a more clear name for the variable.